### DISCUSSION TITLE
Group non-admission add-ons in backend

### DISCUSSION THREAD

**Discussion by vb-03** [pretix/pretix#4712] [Ideas]
Hi,
When a user orders multiple times the same add-on, it is neatly grouped on the user page

![image](https://github.com/user-attachments/assets/9569e24c-7b9d-4253-9402-cfe2eda78310)

However, in back-end, it is separated - which makes reading a user order harder with events with multiple add-ons
![image](https://github.com/user-attachments/assets/743534f5-4fc2-44f4-80c2-7b876ab7c17f)

Product configuration:
![image](https://github.com/user-attachments/assets/ba02ad39-c194-4aae-805a-6f5d06b9911c)

**Comment by raphaelm** [comment]
Hmm, the problem is that in the backend, there are buttons (here one, possibly more) for every product, so collapsing them has some challenges

**Reply by vb-03** [reply]
Indeed - it seems despite this being a non-personalized, not ticket-generating add-on, it generates the ticket Code and a button for that.
Maybe only allowing collapsing if this is set to never, therefore hiding that button for those types of products, is a viable option here?
![image](https://github.com/user-attachments/assets/5ff27b7e-df35-4ffd-ba1d-d4a9e1075043)

**Reply by raphaelm** [reply]
>  Maybe only allowing collapsing if this is set to never, therefore hiding that button for those types of products, is a viable option here?

"Generating tickets" and "having a barcode" is not really equivalent, e.g. when badges are used. The problem is more general. Plugins can (and do) add more buttons to the list. So we'd need logic to only group the lines if no button was rendered, which is complex logic for the very rare use case of 0 buttons. :(
